stix2validator/validator.py

```diff
@@ -341,28 +393,12 @@ def validate_string(string, options=None):
         options: An instance of ``ValidationOptions``.
 
     Returns:
-        An instance of ValidationResults.
+        An instance of ObjectValidationResults.
 
     """
-    results = ValidationResults(fn="input string")
     output.info("Performing JSON schema validation on input string: " + string)
-    instance = json.loads(string)
-
-    if not options:
-        options = ValidationOptions()
-
-    try:
-        results = validate_instance(instance, options)
-    except SchemaInvalidError as ex:
-        results.fatal = ValidationErrorResults(ex)
-        msg = ("String was schema-invalid. No further validation "
-               "will be performed.")
-        output.info(msg.format(string=string))
-
-    if results.errors or results.fatal:
-        results.is_valid = False
-
-    return results
+    stream = io.StringIO(string)
```

Comment:
Well I hadn't realized that python2's `io.StringIO` and `StringIO.StringIO` behaved differently (which is what `six.StringIO` [switches between](http://pythonhosted.org/six/#six.StringIO)).  Ughh...  On the other hand, you've probably seen my other posts harping on python2's confusion of text and binary data, and insisting that `unicode` was the text type and `str` was the binary type.  Insisting on `str` for textual JSON feels a bit like not practicing what I preach... :)

I feel like `StringIO.StringIO` is very legacy; it no longer exists in python3, and many aspects of this library involve unicode whether you like it or not.  E.g. `json.loads` produces unicode keys and values.  In code that we control, perhaps we should prefer modern, forward-looking practices instead of using old APIs.  Both python2 and 3 have `io.StringIO`, so if you're dealing with text properly (as unicode), that particular compatibility feature of `six` isn't necessary, at least on the versions of Python we are supporting.

Reply:
A common practice I'd recommend is to convert from bytes into Unicode as early as possible (when reading bytes from the network or a file, in cases when the `json` library doesn't do that automatically), and back to bytes as late as possible, and assume/enforce that everywhere in the application handles just Unicode text. If six makes that easier, let's do that. I'd also rather use the [`unicode_literals` `__future__` import](https://docs.python.org/3/library/__future__.html) where that makes sense.